### PR TITLE
sneakbox keymap and userspace code (redo)

### PR DESCRIPTION
**Pull request by stanrc85**
Modified userspace files to work with Sneakbox Alice PCB.

Trying again since I messed up my last PR #10954
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: users/stanrc85/indicator_layers_sneakbox.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include "stanrc85.h"
```
Comment (drashna) [inline users/stanrc85/indicator_layers_sneakbox.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/stanrc85/config.h [hunk 2]
```diff
@@ -7,3 +7,9 @@
   #define RGBLIGHT_LAYERS
   #define RGBLIGHT_SLEEP
 #endif
+
+#ifdef SNEAKBOX
+  #define INDICATOR_PIN_0 D7
+  #define INDICATOR_PIN_1 D6
+  #define INDICATOR_PIN_2 D4 
+#endif
```
Comment (drashna) [inline users/stanrc85/config.h:15]: You could just do this: https://docs.qmk.fm/#/feature_led_indicators?id=configuration-options

And you wouldn't need any extra code in your userspace, either.

Comment (stanrc85) [reply users/stanrc85/config.h:15]: I'm not using the indicators for num lock or scroll lock though so I wasn't sure that helped me, unless I'm misunderstanding.

File: users/stanrc85/indicator_layers_sneakbox.c (hunk 1 above)
Comment (stanrc85) [reply users/stanrc85/indicator_layers_sneakbox.c:1]: Added

**Review by drashna — APPROVED** [review]

File: users/stanrc85/indicator_layers_sneakbox.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+    return state;
+}
+
+bool led_update_user(led_t led_state) {
+    writePin(INDICATOR_PIN_0, led_state.caps_lock);
+    writePin(INDICATOR_PIN_1, !middle);
+    writePin(INDICATOR_PIN_2, !bottom);
+    return false;
+}
```
Comment (drashna) [inline users/stanrc85/indicator_layers_sneakbox.c:46]: You could do something like this, instead: 
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(INDICATOR_PIN_2, layer_state_cmp(state, _FN1_60));
    writePin(INDICATOR_PIN_1, layer_state_cmp(state, _FN2_60));
    
    return state;
}

bool led_update_user(led_t led_state) {
    writePin(INDICATOR_PIN_0, led_state.caps_lock);
    return false;
}
```

Comment (stanrc85) [reply users/stanrc85/indicator_layers_sneakbox.c:46]: I like how clean that is, but with that only Caps Lock indicator works.  Is there something else it needs?

Comment (fauxpark) [reply users/stanrc85/indicator_layers_sneakbox.c:46]: The `layer_state_cmp()`s may need to be negated?

Comment (stanrc85) [reply users/stanrc85/indicator_layers_sneakbox.c:46]: layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(INDICATOR_PIN_2, layer_state_cmp(state, _FN2_60));
    writePin(INDICATOR_PIN_1, layer_state_cmp(state, _FN1_60));
    writePin(INDICATOR_PIN_1, layer_state_cmp(state, _DEFAULT));
    writePin(INDICATOR_PIN_2, layer_state_cmp(state, _DEFAULT));
    
    return state;
}
    
If I use that code then it only works when on the _DEFAULT layer, but the code Drashna suggested does work but its missing my two indicators for _DEFAULT.  Looks like it doesn't want to worth both ways?

Comment (drashna) [reply users/stanrc85/indicator_layers_sneakbox.c:46]: You could use `(layer_state_cmp(state, _FN2_60) ||  layer_state_cmp(state, _DEFAULT))` That may work.

Comment (stanrc85) [reply users/stanrc85/indicator_layers_sneakbox.c]: That worked, thank you!

**Review by fauxpark — APPROVED** [review]

**Pull request merged**